inc/cloudfront_media_purge/namespace.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Altis\Cloud\Cloudfront_Media_Purge;
+
+use Exception;
+use function Altis\get_aws_sdk;
+
+function bootstrap() {
+	add_action( 'delete_attachment', __NAMESPACE__ . '\\purge_media_file_cache', 100 );
+}
+
+/**
+ * Return an AWS CloudFront Client instance
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_aws_client() {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => 'latest',
+	] );
+}
+
+/**
+ * Purge an attachment file cache from CloudFront
+ *
+ * It sends a request to invalidate all sizes given for a media file
+ *
+ * @param int $media_id Attachment ID.
+ *
+ * @return bool
+ */
+function purge_media_file_cache( $media_id ) {
+	$meta = wp_get_attachment_metadata( $media_id );
+	if ( ! $meta ) {
+		return false;
+	}
+	$baseurl = '/tachyon/';
+	$basedir = trailingslashit( dirname( $meta['file'] ) );
+	$items   = [ $baseurl . $meta['file'] . '*' ];
+
+	if ( ! empty( $meta['sizes'] ) ) {
+		foreach ( $meta['sizes'] as $size => $size_meta ) {
+			$items[] = $baseurl . $basedir . $size_meta['file'] . '*';
+		}
+	}
+
+	$client = get_aws_client();
```

Comment:
Could we introduce a helper function in `Altis\Cloud` that is something like "purge_cdn_path( string $path_pattern )" or something? Thinkink it would be good to provide that function to custom code too. Then use that function here too.

Reply:
Yes, that make sense. We can create a separate helper function which can create invalidation for given string pattern paths.